picby_graphql_api/src/modules/user/register/isEmailAlreadyExists.ts

```diff
@@ -0,0 +1,25 @@
+import { registerDecorator, ValidationOptions, ValidatorConstraint, ValidatorConstraintInterface } from 'class-validator';
+import { User } from '../../../entity/User';
+
+@ValidatorConstraint({ async: true })
+export class IsEmailAlreadyExistConstraint
+  implements ValidatorConstraintInterface {
+  validate(email: string) {
+    return User.findOne({where:{email}}).then(user => {
+      if (user) return false;
```

Comment:
Just a suggestion but I'd add braces to this if or at least break the line - way less error prone if we stick to it.

Reply:
or ternary should work here :)